src/test/java/ResultsServletTest.java

```diff
@@ -0,0 +1,180 @@
+      results from Datastore based on user keyword input).
+      Not directly mocking the object because 1) Datastore is a final class
+      and 2) The following test ensures the same logic holds while using
+      the existing mocked Datastore service (testDatastore). */
+  @Test
+  public void doGetContentTest_singleKeyword() throws Exception {
+    List<String> eggQuery = new ArrayList<String>();
+    eggQuery.add("EGG");
+    Filter containsEgg = new FilterPredicate("search-strings", FilterOperator.IN, eggQuery);
+
+    Assert.assertEquals(2, testDatastore.prepare(new Query("Recipe").setFilter(containsEgg)).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void doGetContentTest_multipleKeywords() throws Exception {
+    List<String> eggQuery = new ArrayList<String>();
+    eggQuery.add("EGG");
+    List<String> butterQuery = new ArrayList<String>();
+    butterQuery.add("BUTTER");
+    Filter containsEggAndButter = new CompositeFilter(CompositeFilterOperator.AND, Arrays.<Filter>asList(
+                                      new FilterPredicate("search-strings", FilterOperator.IN, eggQuery),
+                                      new FilterPredicate("search-strings", FilterOperator.IN, butterQuery)));
+
+    Assert.assertEquals(1, testDatastore.prepare(new Query("Recipe").setFilter(containsEggAndButter)).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void formatQueryAsListTest_multipleKeywords() {
+    List<String> expectedList = new ArrayList<String>();
+    expectedList.add("EGG");
+    expectedList.add("FLOUR");
+    expectedList.add("BREAD");
+    expectedList.add("BUTTER");
+
+    // formatQueryAsList should be able to correctly separate keywords
+    // despite capitalization and whitespace quirks, so long as the keywords
+    // are separated in some way by commas.
+    String query = "EGG, Flour,bread,BUtter";
+    List<String> actualList = ResultsServlet.formatQueryAsList(query);
+
+    Assert.assertEquals(expectedList, actualList);
+  }
+
+  @Test
+  public void generateFiltersFromQueryTest_noKeywords() {
+    List<String> queryList = new ArrayList<String>();
+    queryList.add("");
+    Filter expectedFilter = new FilterPredicate("search-strings", FilterOperator.IN, queryList);
+
+    Filter actualFilter = ResultsServlet.generateFiltersFromQuery(queryList);
+
+    Assert.assertEquals(expectedFilter, actualFilter);
+  }
+
+  @Test
+  public void generateFiltersFromQueryTest_singleKeyword() {
+    List<String> queryList = new ArrayList<String>();
+    queryList.add("BREAD");
+    Filter expectedFilter = new FilterPredicate("search-strings", FilterOperator.IN, queryList);
+
+    Filter actualFilter = ResultsServlet.generateFiltersFromQuery(queryList);
+    
+    Assert.assertEquals(expectedFilter, actualFilter);
+  }
+
+  @Test
+  public void generateFiltersFromQueryTest_multipleKeywords() {
+    List<String> breadQuery = new ArrayList<String>();
```

Comment:
Same as above regarding iterable for value.

Reply:
Addressed above.